### PR TITLE
Global reporting

### PR DESCRIPTION
**Pull request by renzodgc**
Added a reporting that condenses data from all areas and sources, weekly and daily

### CODE REVIEW

File: config-x86-gpu.ini [hunk 1]
```diff
@@ -25,6 +25,10 @@ ScreenshotsDirectory = /repo/data/processor/static/screenshots
 SlackChannel = lanthorn-notifications
 OccupancyAlertsMinInterval = 180
 MaxThreadRestarts = 5
+GlobalReportingEmails = renzodgc12@gmail.com,renzo@xmartlabs.com
```
Comment (pgrill) [inline config-x86-gpu.ini:28]:
```suggestion
GlobalReportingEmails =
```

File: config-x86-gpu.ini [hunk 2]
```diff
@@ -25,6 +25,10 @@ ScreenshotsDirectory = /repo/data/processor/static/screenshots
...
+GlobalReportTime = 06:00
+DailyGlobalReport = True
```
Comment (pgrill) [inline config-x86-gpu.ini:30]:
```suggestion
DailyGlobalReport = False
```

File: config-x86-gpu.ini [hunk 3]
```diff
@@ -25,6 +25,10 @@ ScreenshotsDirectory = /repo/data/processor/static/screenshots
...
+WeeklyGlobalReport = True
```
Comment (pgrill) [inline config-x86-gpu.ini:31]:
```suggestion
WeeklyGlobalReport = False
```

File: libs/config_engine.py [hunk 4]
```diff
@@ -196,7 +197,7 @@ def get_areas(self):
 
     def should_send_email_notifications(self, entity):
         if "emails" in entity:
-            if os.path.isfile("oauth2_cred.json"):
+            if is_email_configured():
```
Comment (pgrill) [inline libs/config_engine.py:200]: I think that it should be:

```suggestion
            if is_mailing_configured():
```

File: config-coral.ini [hunk 5]
```diff
@@ -12,7 +12,10 @@ SlackChannel = lanthorn-notifications
 ; OccupancyAlertsMinInterval time is measured in seconds (if interval < 0 then no occupancy alerts are triggered)
 OccupancyAlertsMinInterval = 180
 MaxThreadRestarts = 5
-
+GlobalReportingEmails =
+GlobalReportTime=06:00
```
Comment (mats-claassen) [inline config-coral.ini:16]:
```suggestion
GlobalReportTime = 06:00
```

File: api/config.py [hunk 6]
```diff
@@ -87,3 +98,18 @@ async def get_processor_info():
...
+
+@config_router.get("/global_report_emails", response_model=GlobalReportingEmailsInfo)
+async def get_report_emails():
+    return {
+        "emails": extract_config()["App"]["GlobalReportingEmails"]
+    }
+
+
+@config_router.put("/global_report_emails")
+async def get_report_emails(emails: GlobalReportingEmailsInfo, reboot_processor: Optional[bool] = True):
```
Comment (mats-claassen) [inline api/config.py:111]: This function should probably be called 'set...' or 'update...', right?

File: config-x86.ini [hunk 7]
```diff
@@ -26,6 +26,10 @@ SlackChannel = lanthorn-notifications
 ; OccupancyAlertsMinInterval time is measured in seconds (if interval < 0 then no occupancy alerts are triggered)
 OccupancyAlertsMinInterval = 180
 MaxThreadRestarts = 5
+GlobalReportingEmails =
+GlobalReportTime = 06:00
+DailyGlobalReport = False
+WeeklyGlobalReport = False
```
Comment (mats-claassen) [inline config-x86.ini:32]: Please also update the config for tensorrt-gpu

File: api/config.py (hunk 6 above)
Comment (renzodgc) [reply api/config.py:111]: :sweat_smile:

File: README.md [hunk 8]
```diff
@@ -86,13 +86,13 @@ You may hardcode a value rather than using the `timezone.sh` script, such as `US
 
 Please note that the bash script may require permissions to execute `chmod +777 timezone.sh`
 
-**Persisting changes on the config.ini file**
+**Persisting changes files**
 
-Adding the respective `config.ini` as a volume (as for example `-v "$PWD/config-jetson.ini":/repo/config-jetson.ini`). Will allow for syncing changes of said file.
+We recommend adding the proyects folder as a mounted volume (`-v "$PWD":/repo`).
```
Comment (mats-claassen) [inline README.md:91]:
```suggestion
We recommend adding the projects folder as a mounted volume (`-v "$PWD":/repo`).
```

**Review by mats-claassen — APPROVED** [review]

**Review by pgrill — APPROVED** [review]

**Pull request merged**